### PR TITLE
Check manifest

### LINKED ISSUE #1907: Cleanup MANIFEST.in
Currently the `MANIFEST.in` file contains some invalid entries and could use an overall cleanup.

As suggested by @The-Compiler in #1901, we probably could add [`check-manifest`](https://pypi.python.org/pypi/check-manifest) to our `linting` environment and use [pyroma](https://bitbucket.org/regebro/pyroma) too.

### PR DESCRIPTION
**Pull request by nicoddemus**
Uses `check-manifest` to ensure our `MANIFEST.in` is in top-shape.

@hpk42, I removed some scripts from the root which don't seem to be used anymore, would love if you can take a quick look and give your opinion.

@The-Compiler if you could take a look as well it would be super.

Fixes #1907

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -11,22 +11,19 @@ include setup.py
...
 
 graft testing
 graft doc
+prune doc/en/_build
 
 exclude _pytest/impl
 
 graft _pytest/vendored_packages
 
 recursive-exclude * *.pyc *.pyo
```
Comment (The-Compiler) [inline MANIFEST.in]: What about the recursive-excludes added in https://github.com/pytest-dev/pytest/pull/1901/files#diff-97c91a104c431d0c365565d3ac03ac13R24 - should we add them here?

**Comment by hpk42** [comment]
looks good to me to remove the old scripts.  We are missing an easy way to test a new pytest release against released versions of some plugins, don't we?  ``devpi test X`` for packages coming from pypi doesn't work, only for packages uploaded to pypi indexes.  that should change i guess because we could then run tests by a series of "devpi test X" while running from an index which contains the to-be-released version of pytest.

File: MANIFEST.in (hunk 1 above)
Comment (nicoddemus) [inline MANIFEST.in]: Ahh good catch, thanks! Updated.

**Comment by nicoddemus** [comment]
> That should change i guess because we could then run tests by a series of "devpi test X" while running from an index which contains the to-be-released version of pytest.

That sounds neat. I've been using [devpi-cloud-tester](https://github.com/nicoddemus/devpi-cloud-tester) for some time to test releases, it would be simple to just add a bunch of selected pytest plugins to test them together with releases. 

`devpi` has a `--index` option, but I guess that won't solve this because we need to tell devpi to test the package from PyPI, but get the dependencies from the index (when possible) and fall back to PyPI for the others as it usually does anyway.

**Comment by hpk42** [comment]
there is one prooblem with this PR: the "check-manifest" call depends on version control and "devpi test" only installs the packages and does not have .hg/.git etc. directories.  so linting fails with "devpi test".   It seems ``check-manifest`` has no way to go without VCS.  What do we do?

**Comment by nicoddemus** [comment]
Oh you're right. Hmm how about moving the command to a separate env, `manifest`, which is not listed in `tox`'s env-list but is called explicitly by one of the CI services (say Travis)?  This way it won't get executed by `devpi test`, but will run continuously on CI.

**Comment by hpk42** [comment]
On Fri, Nov 11, 2016 at 07:08 -0800, Bruno Oliveira wrote:
> Oh you're right. Hmm how about moving the command to a separate env, `manifest`, which is not listed in `tox`'s env-list but is called explicitly by one of the CI services (say Travis)?  This way it won't get executed by `devpi test`, but will run continuously on CI.

probably works, feels a bit like a workaround.  A more direct workaround might be to call a script a script which calls check-manifest only if there is a .git dir and otherwise prints a "no .git directory found, skipping checking the manifest file" or so.

**Comment by nicoddemus** [comment]
I like your plan better. 😁 

I will open a PR later.

**Pull request merged**